src/modules/esm.mjs

```diff
@@ -0,0 +1,44 @@
+import path, {dirname} from "path";
+import {release, version} from "os";
+import {createServer} from "http";
+import {fileURLToPath} from "url";
+import "./files/c.js";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const random = Math.random();
+
+export const unknownObject = random > 0.5
+  ? await import('./files/a.json', {assert: {type: 'json'}})
```

Comment:
This code produces warnings in the console, which we didn't have with CommonJS before.
You should use `createRequire` instead.